src/nvim/lua/vim.lua

```diff
@@ -123,7 +123,7 @@ function vim._load_package(name)
     local path = "lua/"..trail:gsub('?',basename)
     local found = vim.api.nvim_get_runtime_file(path, false)
     if #found > 0 then
-      return package.loadlib(found[1])
+      return package.loadlib(found[1], "luaopen_" .. name)
```

Comment:
what happens if the so is namespaced like `lua/myfoo/bar.so` ? shouldn't `luaopen_bar` be used then?

Reply:
Good point. I've fixed this and also made the naming more in line with what lua does by default.